File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/ports/ObservationRepositoryPort.kt
package br.gov.educacao.sp.pecobservation.ports

import br.gov.educacao.sp.pecobservation.domain.entities.Observation
import br.gov.educacao.sp.pecobservation.domain.entities.ObservationStatus
import kotlinx.coroutines.flow.Flow
import java.util.UUID

interface ObservationRepositoryPort {
    fun observeAll(): Flow<List<Observation>>
    suspend fun save(observation: Observation)
    suspend fun fetchById(id: UUID): Observation?
    suspend fun fetchByStatus(status: ObservationStatus): List<Observation>
    suspend fun delete(id: UUID)
}

File: pec-observation-app/ios-app/Sources/PECObservation/Ports/ObservationRepositoryPort.swift
import Foundation

public protocol ObservationRepositoryPort {
    func save(_ observation: Observation) throws
    func fetchAll() throws -> [Observation]
    func fetchById(_ id: UUID) throws -> Observation?
    func fetchByStatus(_ status: ObservationStatus) throws -> [Observation]
    func delete(_ id: UUID) throws
}
